sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/ServiceItemLease.java

```diff
@@ -236,4 +235,32 @@ public String toString() {
             this.getTimestamp(),
             UNIX_START_TIME.plusSeconds(Long.parseLong(this.getTs())));
     }
+
+    @SuppressWarnings("serial")
+    static final class ServiceItemLeaseJsonSerializer extends StdSerializer<ServiceItemLease> {
+        // this value should be incremented if changes are made to the ServiceItemLease class members
+        private static final long serialVersionUID = 1L;
+
+        protected ServiceItemLeaseJsonSerializer() { this(null); }
+
+        protected ServiceItemLeaseJsonSerializer(Class<ServiceItemLease> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(ServiceItemLease lease, JsonGenerator writer, SerializerProvider serializerProvider) {
+            try {
+                writer.writeStartObject();
+                writer.writeStringField(Constants.Properties.ID, lease.getId());
+                writer.writeStringField(Constants.Properties.E_TAG, lease.getETag());
+                writer.writeStringField("LeaseToken", lease.getLeaseToken());
+                writer.writeStringField("ContinuationToken", lease.getContinuationToken());
+                writer.writeStringField("timestamp", lease.getTimestamp());
+                writer.writeStringField("Owner", lease.getOwner());
```

Comment:
LeaseToken, ContinuationToken, Owner are Camel case, but timestamp, id, etag are not is that expected?

also for Owner, timestamp, ContinuationToken, LeaseToken we probably should define constants.

Reply:
The current naming convention follows .Net and the content of a lease document as written from that code. The idea behind this design is that you can have CFP instances resuming work from both platforms, Java and .NET.
The ServiceItemLease is internal only (not exposed to the user); for the easy of debugging keeping the same names for members and the content of the lease document as written in Cosmos is actually beneficial.